hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WAL.java

```diff
@@ -125,15 +128,11 @@
    * @param key Modified by this call; we add to it this edits region edit/sequence id.
    * @param edits Edits to append. MAY CONTAIN NO EDITS for case where we want to get an edit
    *          sequence id that is after all currently appended edits.
-   * @param closeRegion Whether this is a region close marker, i.e, the last wal edit for this
-   *          region on this region server. The WAL implementation should remove all the related
-   *          stuff, for example, the sequence id accounting.
    * @return Returns a 'transaction id' and <code>key</code> will have the region edit/sequence id
    *         in it.
    * @see #appendData(RegionInfo, WALKeyImpl, WALEdit)
    */
-  long appendMarker(RegionInfo info, WALKeyImpl key, WALEdit edits, boolean closeRegion)
-    throws IOException;
+  long appendMarker(RegionInfo info, WALKeyImpl key, WALEdit edits) throws IOException;
```

Comment:
Only one append method is enough? In the implementation, we could first test if it is an meta edit, just checking whether there is only one cell and then check the family,, if so the inMemstore is false...

Anyway, the current interface is also fine, maybe it is better for performance...

Reply:
Keeping them for now. If the isMetaEdit test was fast, I'd remove them. While it is ugly, will keep them.